internals/review_models.py

```diff
@@ -188,3 +188,131 @@ def get_raw_owner_file(cls, url):
       return None
 
     return owners_file.raw_content
+
+
+# Note: This class is not used yet.
+class Gate(ndb.Model):  # copy from ApprovalConfig
+  """Gates regulate the completion of a stage."""
+  feature_id = ndb.IntegerProperty(required=True)
+  stage_id = ndb.IntegerProperty(required=True)
+  gate_type = ndb.IntegerProperty(required=True)  # copy from field_id
+
+  # Can be REVIEW_REQUESTED or one of ApprovalValue states
+  state = ndb.IntegerProperty(required=True)  # calc from Approval
+
+  owners = ndb.StringProperty(repeated=True)
+  next_action = ndb.DateProperty()
+  additional_review = ndb.BooleanProperty(default=False)
+
+  # TODO(jrobbins): implement request_review() and clear_request()
+
+
+# Note: This class is not used yet.
+class Vote(ndb.Model):  # copy from Approval
+  """One approver's vote on what the state of a gate should be."""
+
+  # Not used: NEEDS_REVIEW = 0
+  NA = 1
+  REVIEW_REQUESTED = 2
+  REVIEW_STARTED = 3
+  NEED_INFO = 4
+  APPROVED = 5
+  NOT_APPROVED = 6
+  VOTE_VALUES = {
+      # Not used: NEEDS_REVIEW: 'needs_review',
+      NA: 'na',
+      REVIEW_REQUESTED: 'review_requested',
+      REVIEW_STARTED: 'review_started',
+      NEED_INFO: 'need_info',
+      APPROVED: 'approved',
+      NOT_APPROVED: 'not_approved',
+  }
+
+  FINAL_STATES = [NA, APPROVED, NOT_APPROVED]
+
+  feature_id = ndb.IntegerProperty(required=True)
+  gate_id = ndb.IntegerProperty(required=True)
+  state = ndb.IntegerProperty(required=True)
```

Comment:
What's 'state'?

Reply:
State will be one of the enum values listed above.

Reply:
I think I'm getting a bit outside my swim lane here. I beg your indulgence.

'State' is not very domain specific. By that I mean that implementation status and process stage are both states. I asked the question thinking this was one of those and wondering which one. (If I had looked at line 231, I would have figured this out.)